packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.test.ts

```diff
@@ -0,0 +1,76 @@
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  MockedCommerceEngine,
+  buildMockCommerceEngine,
+} from '../../../../test/mock-engine-v2';
+import {buildCorePagination} from '../pagination/headless-core-commerce-pagination';
+import {buildCoreInteractiveResult} from '../result-list/headless-core-interactive-result';
+import {buildCoreSort} from '../sort/headless-core-commerce-sort';
+import {
+  buildSolutionTypeSubControllers,
+  SolutionTypeSubControllers,
+} from './headless-sub-controller';
+
+jest.mock('../result-list/headless-core-interactive-result');
+jest.mock('../pagination/headless-core-commerce-pagination');
+jest.mock('../sort/headless-core-commerce-sort');
+
+describe('sub controllers', () => {
+  let engine: MockedCommerceEngine;
+  let subControllers: SolutionTypeSubControllers;
+  const mockResponseIdSelector = jest.fn();
+  const mockFetchResultsActionCreator = jest.fn();
+
+  function initSubControllers() {
+    engine = buildMockCommerceEngine(buildMockCommerceState());
+
+    subControllers = buildSolutionTypeSubControllers(engine, {
+      responseIdSelector: mockResponseIdSelector,
+      fetchResultsActionCreator: mockFetchResultsActionCreator,
+    });
+  }
+
+  beforeEach(() => {
+    initSubControllers();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('#interactiveResult builds interactive result controller', () => {
+    const props = {
+      options: {
+        product: {
+          productId: '1',
+          name: 'Product name',
+          price: 17.99,
+        },
+        position: 1,
+      },
+    };
+
+    subControllers.interactiveResult({
+      ...props,
+    });
+
+    expect(buildCoreInteractiveResult).toHaveBeenCalledWith(engine, {
```

Comment:
Another way to test this would be to mock the `buildCoreInteractiveResult` function and validate that the value returned by `subControllers.interactiveResult({ ... props });` is equal to the returned value of the `buildCoreInteractiveResult` mock.

E.g.,

```typescript
// ...
// Commenting this to show that it would have to be removed:
// import {buildCoreInteractiveResult} from '../result-list/headless-core-interactive-result';

import * as CoreInteractiveResult from '../result-list/headless-core-interactive-result';

// ...

// Commenting this to show that it would have to be removed:
// jest.mock('../result-list/headless-core-interactive-result');

describe('sub controllers', () => {
  // ...

  it('#interactiveResult builds interactive result controller', () => {
    const buildCoreInteractiveResultMock = jest.spyOn(CoreInteractiveResult, 'buildCoreInteractiveResult');
    
    const props = {
      options: {
        product: {
          productId: '1',
          name: 'Product name',
          price: 17.99,
        },
        position: 1,
      },
    };

    const interactiveResult = subControllers.interactiveResult({
      ...props,
    });

    expect(interactiveResult).toEqual(buildCoreInteractiveResultMock.mock.results[0].value);
  });
});
```

Reply:
Similar comment for other builder tests.

Reply:
Great idea! Applied your suggestion in https://github.com/coveo/ui-kit/pull/3800/commits/86f3881890a7b7e1ca40b7b28938a7f871a0794d